src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,37 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private static final StorageDao storageDao = new StorageDaoImpl();
+    private static final int MIN_AGE = 18;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_LOGIN_LENGTH = 6;
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null || user.getLogin().isEmpty()) {
+            throw new RegistrationException("Login cannot be null or empty");
+        }
+        if (user.getPassword() == null || user.getPassword().isEmpty()) {
+            throw new RegistrationException("Password cannot be null or empty");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_LENGTH) {
+            throw new RegistrationException("Login length cannot be less than " + MIN_LOGIN_LENGTH);
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password length cannot be less than "
+                    + MIN_PASSWORD_LENGTH);
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("User cannot be under " + MIN_AGE);
+        }
+        if (user.getAge() > Integer.MAX_VALUE) {
+            throw new RegistrationException("Age is incorrect");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("User already exist");
+        }
+        storageDao.add(user);
```

Comment:
The method should return the result of the storageDao.add(user) call to ensure that the returned User is the one that was actually stored.